r55/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+pub mod exec;
+
+mod error;
+
+pub mod test_utils;
+
+use std::fs::File;
+use std::io::Read;
+use std::process::Command;
+
+fn compile_runtime(path: &str) -> eyre::Result<Vec<u8>> {
+    println!("Compiling runtime: {}", path);
+    let status = Command::new("cargo")
+        .arg("+nightly-2024-02-01")
+        .arg("build")
+        .arg("-r")
+        .arg("--lib")
+        .arg("-Z")
+        .arg("build-std=core,alloc")
+        .arg("--target")
+        .arg("riscv64imac-unknown-none-elf")
+        .arg("--bin")
+        .arg("runtime")
+        .current_dir(path)
+        .status()
+        .expect("Failed to execute cargo command");
+
+    if !status.success() {
+        eprintln!("Cargo command failed with status: {}", status);
+        std::process::exit(1);
+    } else {
+        println!("Cargo command completed successfully");
+    }
+
+    let path = format!(
+        "{}/target/riscv64imac-unknown-none-elf/release/runtime",
+        path
+    );
+    let mut file = match File::open(path) {
+        Ok(file) => file,
+        Err(e) => {
+            eyre::bail!("Failed to open file: {}", e);
+        }
+    };
+
+    // Read the file contents into a vector.
+    let mut bytecode = Vec::new();
+    if let Err(e) = file.read_to_end(&mut bytecode) {
+        eyre::bail!("Failed to read file: {}", e);
+    }
+
+    Ok(bytecode)
+}
+
+pub fn compile_deploy(path: &str) -> eyre::Result<Vec<u8>> {
+    compile_runtime(path)?;
+    println!("Compiling deploy: {}", path);
```

Comment:
lets set up a logger and use log::info instead of println whenever feasible

Reply:
Agree. Implemented a new method `initialize_logger` and set up logger.